packages/library-base/src/commonMain/kotlin/io/realm/kotlin/internal/RealmUtils.kt

```diff
@@ -165,15 +168,30 @@ internal fun <T : BaseRealmObject> copyToRealm(
             }
         }
         val target = if (hasPrimaryKey) {
-            @Suppress("UNCHECKED_CAST")
-            create(
-                mediator,
-                realmReference,
-                element::class,
-                className,
-                RealmValueArgumentConverter.convertArg(primaryKey),
-                updatePolicy
-            )
+            // Check only for Strings since ByteArrays cannot be primary keys
+            if (primaryKey is String) {
```

Comment:
Not sure it is worth differentiating ... and maybe not possible to differentiate in the type system if we get the receiver type of `convertArg` right ;)